specifyweb/frontend/js_src/lib/components/SchemaConfig/Tables.tsx

```diff
@@ -85,20 +85,28 @@ export function TableList({
   );
 
   const sortedTables = React.useMemo(
+    () => Object.values(tables).sort(sortFunction(({ name }) => name)),
+    []
+  );
+
+  const presentFilteredTables = React.useMemo(
     () =>
-      Object.values(tables)
-        .filter(
-          filter?.bind(undefined, showHiddenTables) ??
-            (showHiddenTables ? f.true : ({ isSystem }): boolean => !isSystem)
-        )
-        .sort(sortFunction(({ name }) => name)),
-    [filter, showHiddenTables]
+      Object.values(sortedTables).filter((table) => {
+        const childrenResult = children?.(table);
+        return typeof childrenResult === 'string';
+      }),
+    [sortedTables, children]
+  );
+
+  const tablesToDisplay = React.useMemo(
+    () => (showHiddenTables ? sortedTables : presentFilteredTables),
+    [showHiddenTables, sortedTables, presentFilteredTables]
```

Comment:
(optional) this can be combined into a single useMemo - would be less code

performance difference won't be noticeable either way

Reply:
initially we used a filter prop for 'advance tables', we do not use it here anymore. Is there any advantages of keep that? At the moment it just displays all tables that already have a resource and when checking the check box all the other ones